### DISCUSSION TITLE
Suggestions produce lines, that already exist

### DISCUSSION THREAD

**Discussion by devidw** [community/community#13281] [Copilot Conversations]
![Screenshot 2022-03-22 at 9 28 44 PM](https://user-images.githubusercontent.com/68775926/159570373-754c87c5-2cbd-4d14-8a2e-d51a57e860c1.png)

Sometimes I get suggestions, where the _last line_ or _the last few lines_ of the suggestion are already coded out.
Some kind of comparing mechanisms, that compares the produced lines to the existing, following lines, would be awesome.

**Comment by Bloodiko** [comment]
yes, this is behavior i also noticed. 

It would be much better if copilot would not only check for the "Previous Lines" for Context, but also the already following ones. 
It often happens that i want to add something or replace something in a method 
and copilot suggests the correct things, but sometimes it wants to add the already existing lines below as well again. 

this happening on bigger functions is kinda okay, but when it happens on completing an 
`if () {}`
while writing in the ( ) brackets - Copilot wants to add { } brackets, (sometimes with actions, sometimes without) again even though they are already there. 
This applies to ` return ` Statements often as well.
